packages/atclient/include/atclient/metadata.h

```diff
@@ -2,60 +2,363 @@
 #ifndef ATCLIENT_METADATA_H
 #define ATCLIENT_METADATA_H
 
+#include "atclient/atsign.h"
 #include "atclient/atstr.h"
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
 
-#define DATE_STR_BUFFER_LENGTH 256 // can hold most date strings found in metadata
-#define GENERAL_BUFFER_LENGTH 8192 // can hold most things found in metadata
+// initializedfields[0]
+#define ATKEY_METADATA_CREATEDBY_INITIALIZED 0b00000001
+#define ATKEY_METADATA_UPDATEDBY_INITIALIZED 0b00000010
+#define ATKEY_METADATA_STATUS_INITIALIZED 0b00000100
+#define ATKEY_METADATA_VERSION_INITIALIZED 0b00001000
+#define ATKEY_METADATA_EXPIRESAT_INITIALIZED 0b00010000
+#define ATKEY_METADATA_AVAILABLEAT_INITIALIZED 0b00100000
+#define ATKEY_METADATA_REFRESHAT_INITIALIZED 0b01000000
+#define ATKEY_METADATA_CREATEDAT_INITIALIZED 0b10000000
```

Comment:
would be cleaner to use bit shift syntax next time (1 << 0), (1 << 1), etc.. to help avoid small typos